### ISSUE TITLE
Question: How to run system & idempotence tests

### ISSUE THREAD

**Issue by calebcartwright** [rust-lang/rustfmt#3627]
Not sure if this is a user error on my part or a bug/issue with these tests not running.

When I run `cargo test` in my local env, the system and idempotence tests don't appear to be running. Relevant snippets from the output:
```sh
Ran 0 idempotent tests.
test test::idempotence_tests ... ok
.....
Ran 0 system tests.
test test::system_tests ... ok
```

This is also the case in the latest builds on Travis and AppVeyor. For example 
https://travis-ci.com/rust-lang/rustfmt/jobs/207595534
https://ci.appveyor.com/project/rust-lang-libs/rustfmt/build/job/il5kok93xmvcaafu
Relevant snippets from the output:
```sh
Ran 0 idempotent tests.
test test::idempotence_tests ... ok
.....
Ran 0 system tests.
test test::system_tests ... ok
```

However, it seems these tests only started being skipped in Travis and AppVeyor ~1 week ago. Before, it seems that both the system and idempotence tests were always running on each job in both Travis and AppVeyor.

For example 
https://travis-ci.com/rust-lang/rustfmt/jobs/206494618
https://ci.appveyor.com/project/rust-lang-libs/rustfmt/builds/25135606/job/fhkd2ac6bin3x88p
Relevant snippets from the output:
```sh
Ran 473 idempotent tests.
test test::idempotence_tests ... ok
...
Ran 372 system tests.
test test::system_tests ... ok
```

Based on the [contributing guide](https://github.com/rust-lang/rustfmt/blob/master/Contributing.md#create-test-cases) I would have expected both the system and idempotence tests to be executed as part of `cargo test`. Is that no longer the case or is there a bug in the discovery/execution of those tests?

**Comment by calebcartwright** [comment]
My sense is that the `is_file_skip`/`is_subpath` check [here](https://github.com/rust-lang/rustfmt/blob/master/src/test/mod.rs#L73) is resulting in every test file being skipped based on the values defined in the [SKIP_FILE_WHITE_LIST](https://github.com/rust-lang/rustfmt/blob/master/src/test/mod.rs#L23), as I _can_ get the tests to run by commenting out the entries in that list

**Comment by topecongiro** [comment]
@calebcartwright Thank you for catching this. This should be fixed with #3628.

**Comment by calebcartwright** [comment]
Awesome!

I had just managed to get it working locally by tweaking the `is_subpath` check to 😄 
```rust
    (0..path.components().count())
        .map(|i| path.components().skip(i).take(subpath.as_ref().components().count()))
        .any(|c| c.zip(subpath.as_ref().components()).all(|(a, b)| a == b))
```

Thanks for getting this resolved so quickly!
